org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalRequestor.java

```diff
@@ -184,33 +187,87 @@ public void accept(CompletionProposal proposal) {
 	}
 
 	public List<CompletionItem> getCompletionItems() {
-		//Sort the results by relevance 1st
+		CompletionRankingAggregation[] aggregatedRanks = getAggregatedRankingResult();
+		for (int i = 0; i < proposals.size(); i++) {
+			CompletionProposal proposal = proposals.get(i);
+			if (aggregatedRanks[i] != null) {
+				// we assume there won't be overflow for now since the the score from
+				// each provider can only be 100 at most.
+				proposal.setRelevance(proposal.getRelevance() + aggregatedRanks[i].getScore());
+			}
+		}
+		Map<CompletionProposal, CompletionRankingAggregation> proposalToRankingResult = new HashMap<>();
+		for (int i = 0; i < proposals.size(); i++) {
+			proposalToRankingResult.put(proposals.get(i), aggregatedRanks[i]);
+		}
+
 		proposals.sort(new ProposalComparator(proposals.size()));
-		List<CompletionItem> completionItems = new ArrayList<>(proposals.size());
 		int maxCompletions = preferenceManager.getPreferences().getMaxCompletionResults();
 		int limit = Math.min(proposals.size(), maxCompletions);
-		if (proposals.size() > maxCompletions) {
-			//we keep receiving completions past our capacity so that makes the whole result incomplete
-			isComplete = false;
-			response.setProposals(proposals.subList(0, limit));
-		} else {
-			response.setProposals(proposals);
-		}
-		CompletionResponses.store(response);
+		List<CompletionItem> completionItems = new ArrayList<>(limit);
 
 		//Let's compute replacement texts for the most relevant results only
 		for (int i = 0; i < limit; i++) {
 			CompletionProposal proposal = proposals.get(i);
 			try {
 				CompletionItem item = toCompletionItem(proposal, i);
+				CompletionRankingAggregation rankingResult = proposalToRankingResult.get(proposal);
+				if (rankingResult != null) {
+					String decorators = rankingResult.getDecorators();
+					if (!decorators.isEmpty()) {
+						item.setLabel(decorators + " " + item.getLabel());
```

Comment:
<img width="521" alt="image" src="https://user-images.githubusercontent.com/2351748/202076807-2aaf2efa-170f-4ffa-8560-af7a602e8a2d.png">

Something wrong. BTW I didn't call setDecorator() for any result, as you should allow that.

Reply:
I need to update the implementation of `CompletionRankingAggregation#getDecorators()` to

```
return this.decorators.stream().sorted().map(String::valueOf).collect(Collectors.joining());
```